preDownloadMultipleSecureFiles.ts

```diff
@@ -0,0 +1,50 @@
+import path = require('path');
+import tl = require('azure-pipelines-task-lib/task');
+import secureFilesCommon = require('securefiles-common/securefiles-common');
+
+const commaDelimiter: string = ',';
+const newLineDelimiter: string = '\n';
+
+const main = async () => {
+  try {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+    let retryCount: number = parseInt(tl.getInput('retryCount') as string);
+
+    if (isNaN(retryCount) || retryCount < 0) {
+      console.log('yes we have nan');
+      retryCount = 5;
+    }
+
+    const secureFiles: string[] = getSecureFiles();
+    const secureFilesPath: string[] = await downloadSecureFiles(secureFiles, retryCount);
+    const existingSecureFilePath: string[] = getExistingSecureFile(secureFilesPath);
+    const secureFilePathsOutput: string = getSecureFilePathsOutput(existingSecureFilePath);
+
+    tl.setVariable('secureFilePaths', secureFilePathsOutput);
+  } catch (err) {
+    tl.setResult(tl.TaskResult.Failed, err);
+  }
+};
+
+const getSecureFiles = (): Array<string> => {
+  let secureFiles: Array<string> = tl.getDelimitedInput('secureFiles', newLineDelimiter, true);
+
+  if (secureFiles !== null && (secureFiles.length <= 0 || (secureFiles.length == 1 && secureFiles[0].includes(commaDelimiter)))) {
+    secureFiles = tl.getDelimitedInput('secureFiles', commaDelimiter, true);
```

Comment:
Why we have this commaDelimiter here? In which case scenario it'll be containing commas?

Reply:
And do we have a test case scenario for it? 🤔

Reply:
I did it because of yaml pipeline and secure variables -> we cannot add there new line, so decided to add it via comma separated values.

Tests for this you can check here: 
 * https://github.com/aochmann/azure-download-multiple-secure-files/blob/e17a7dcd2ec2c2ef42e22ead8ad95c53f9b0e5fb/Tests/L0.ts#L59
 * https://github.com/aochmann/azure-download-multiple-secure-files/blob/e17a7dcd2ec2c2ef42e22ead8ad95c53f9b0e5fb/Tests/L0MultipleFilesCommaDelimiter.ts